hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapEntryView.java

```diff
@@ -0,0 +1,252 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.wan;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.nio.IOUtil;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.BinaryInterface;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+
+/**
+ * WAN heap based implementation of {@link EntryView} for keeping
+ * compatibility when sending to older (3.8+) clusters.
+ *
+ * @param <K> the type of key.
+ * @param <V> the type of value.
+ */
+@BinaryInterface
+public class WanMapEntryView<K, V> implements EntryView<K, V>, IdentifiedDataSerializable {
+
+    private final K key;
+    private final V value;
+    private final long cost;
+    private final long creationTime;
+    private final long expirationTime;
+    private final long hits;
+    private final long lastAccessTime;
+    private final long lastStoredTime;
+    private final long lastUpdateTime;
+    private final long version;
+    private final long ttl;
+
+    public WanMapEntryView(EntryView<K, V> entryView) {
+        this.key = entryView.getKey();
+        this.value = entryView.getValue();
+        this.cost = entryView.getCost();
+        this.version = entryView.getVersion();
+        this.hits = entryView.getHits();
+        this.lastAccessTime = entryView.getLastAccessTime();
+        this.lastUpdateTime = entryView.getLastUpdateTime();
+        this.ttl = entryView.getTtl();
+        this.creationTime = entryView.getCreationTime();
+        this.expirationTime = entryView.getExpirationTime();
+        this.lastStoredTime = entryView.getLastStoredTime();
+    }
+
+    @Override
+    public K getKey() {
+        return key;
+    }
+
+    @Override
+    public V getValue() {
+        return value;
+    }
+
+    @Override
+    public long getCost() {
+        return cost;
+    }
+
+    @Override
+    public long getCreationTime() {
+        return creationTime;
+    }
+
+    @Override
+    public long getExpirationTime() {
+        return expirationTime;
+    }
+
+    @Override
+    public long getHits() {
+        return hits;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return lastAccessTime;
+    }
+
+    @Override
+    public long getLastStoredTime() {
+        return lastStoredTime;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    @Override
+    public long getVersion() {
+        return version;
+    }
+
+    @Override
+    public long getTtl() {
+        return ttl;
+    }
+
+    @Override
+    public long getMaxIdle() {
+        return 0;
```

Comment:
Should this maybe return `RecordStore.DEFAULT_TTL` (-1 --> use `MapConfig`'s default) instead of `0` (--> infinite max idle)?

Reply:
Looks like there's some issue at play with [this fix](https://github.com/hazelcast/hazelcast/pull/13632). Once the source cluster sends `-1`, the `maxIdle` of one second is put to the target cluster record:
https://github.com/mmedenjak/hazelcast/blob/b509799e39b7ca56f111a1a84c6457c51b159a9a/hazelcast/src/main/java/com/hazelcast/map/impl/record/AbstractRecord.java#L91-L94

I'm not quite sure how this RU fix is supposed to work and how to make it compatible with the WAN entry merging. Maybe @tkountis can help out here.

Reply:
I'm going to take a wild guess here and say that `com.hazelcast.core.EntryView#getMaxIdle` should return a boxed `Long`, whereas `WanMapEntryView#getMaxIdle` would return `null` as it does not carry any `maxIdle` information.

Reply:
@tkountis can you review this PR? It tries to work with the max idle RU fix, not sure if it's correct.

Reply:
Sure